osu.Game/Rulesets/Objects/Pooling/PoolableDrawableWithLifetime.cs

```diff
@@ -65,7 +84,7 @@ protected override void LoadAsyncComplete()
         {
             base.LoadAsyncComplete();
 
-            // Apply the initial entry given in the constructor.
+            // Apply the initial entry.
             if (Entry != null && !HasEntryApplied)
                 Apply(Entry);
```

Comment:
One thing to keep in mind is this can be race-y. This code is running on the async thread while `LoadState = Loading`, so it's possible that this can apply a previous entry, then a new one comes in via `Entry_set`, gets stored to `entry` but never actually gets applied.

I'm not sure how big of a deal this is, but perhaps easiest is to change to
```
if (LoadState == LoadState.NotLoaded)
    entry = value;
else if (value != null)
{
    if (LoadState < LoadState.Ready)
        Schedule(() => Apply(value));
    else
        Apply(value);
}
else if (HasEntryApplied)
    free();
```
I don't necessarily care much for performance, but you could cancel the scheduled delegate in `Entry_set` if you want.

Reply:
Do you think it is okay to just disallow (throws exception) setting entry while loading?

Reply:
Yup, that'd work too.

Reply:
If the request for review here was about this point then I'm also fine with throwing when this is called during load. That said, I'm not necessarily sure that `Apply()` on its own is free of this same sort of race itself...